Fix the issue of Mono.AotCross build errors

Linlccc opened the pull request:
Configure the DotNetBuildMonoCrossAOT property when building Mono.AotCross

Fixes dotnet#117970

Linlccc committed:
Fix the issue of Mono.AotCross build errors
Configure the DotNetBuildMonoCrossAOT property when building Mono.AotCross

Fixes dotnet#117970

akoeplinger commented:
if you just build mono.aotcross the expectation is that you set `MonoCrossAOTTargetOS`

Linlccc committed:
Fix the issue of Mono.AotCross build errors
Remove the DotNetBuildMonoCrossAOT default value when building Mono.AotCross
Check MonoAotTarget before building and prompt

Fixes #117970

akoeplinger commented:
so is running AllSubset your main use case? this was added basically just for restoring: https://github.com/dotnet/runtime/pull/117374

I'm not sure it makes much sense to support this for building too

Linlccc commented:
> so is running AllSubsets your main use case? this was added basically just for restoring: #117374
> 
> I'm not sure it makes much sense to support this for building too

Yes, now it will only check the target and give a prompt before building

Linlccc commented:
@akoeplinger Now, is there any problem with this?

steveisok reviewed (approved):
Looks fine to me.
